v3/src/components/graph/models/graph-data-configuration-model.ts

```diff
@@ -559,7 +559,10 @@ export const GraphDataConfigurationModel = DataConfigurationModel
           return isBottomMatch && isTopMatch && isRightMatch
         })
       }
-    })
+    }),
+    filteredForShowMeasuresForSelection(caseIds: string[]) {
+      return self.showMeasuresForSelection ? caseIds.filter(caseId => self.dataset?.isCaseSelected(caseId)) : caseIds
```

Comment:
1. Since this function is doing the filtering and there may at some point be other reasons for filtering cases, I'd call it something like `filterCasesForDisplay`.
2. Since many clients end up having to call this function like this: `filterCasesForDisplay(config?.getCases() ?? [])`, you could simplify their lives by accepting `undefined` and returning an empty array.
```suggestion
    filterCasesForDisplay(caseIds: string[] = []) {
      return self.showMeasuresForSelection ? caseIds.filter(caseId => self.dataset?.isCaseSelected(caseId)) : caseIds
```